server/src/features/orders/controller.ts

```diff
@@ -123,103 +120,120 @@ export class ProductOrdersController extends Controller {
+              type: "fixed_amount",
+              fixed_amount: {
+                amount: 1000,
+                currency: "eur",
+              },
+              display_name: "Next day shipping",
+              delivery_estimate: {
+                minimum: {
+                  unit: "business_day",
+                  value: 1,
+                },
+                maximum: {
+                  unit: "business_day",
+                  value: 1,
+                },
+              },
+            },
+          },
+        ],
+        metadata: {
+          userId: user.id,
+        },
       });
       req.res?.json({ url: session.url });
     } catch (e: any) {
       req.res?.status(500).json({ error: e.message });
     }
   }
-}
-const prisma = new PrismaClient();
-const endpointSecret = ENDPOINT_SECRET;
-
-@Route("/webhook")
-@Tags("Orders")
-export class WebhookController extends Controller {
-  @Post()
-public async createWebhook(
-  @Request() req: ExRequest,
-): Promise<void>{
-  let event = req.body;
-    // Only verify the event if you have an endpoint secret defined.
-    // Otherwise use the basic event deserialized with JSON.parse
-    if (endpointSecret) {
-      // Get the signature sent by Stripe
-      const signature = req.headers["stripe-signature"];
-      try {
-        event = stripe.webhooks.constructEvent(
-          req.body.text(),
-          signature!,
-          endpointSecret
-        );
-      } catch (err: any) {
-        console.log(`⚠️  Webhook signature verification failed.`, err.message);
-      }
-    }
-    // Handle the event
+  /**
+   * Creates Stripe webhook to update database is the checkout session was successfull.
+   */
+  @Post("/webhook")
+  public async createWebhook(@Request() req: ExRequest): Promise<void> {
+    let event = req.body;
+    const metadata = event.data.object;
     switch (event.type) {
       case "checkout.session.completed":
-        const checkoutSessionCompleted = event.data.object;
-        //Then define and call a function to handle the event checkout.session.completed
-        const user = await prisma.user.findUnique({
-          where: {
-            email: event.data.object.customer_details.email,
-          },
-        });
-
-        const prismaOrder = await prisma.order.findFirst({
-          where: {
-            userId: user!.id,
-          },
-          orderBy: {
-            id: "desc",
-          },
-        });
-
-        await prisma.order.update({
-          where: {
-            id: prismaOrder!.id,
-          },
-          data: {
-            status: "Payment Received",
-            amount: event.data.object.amount_total / 100,
-          },
-        });
-        console.log(checkoutSessionCompleted);
+        const prismaOrder = await new OrderService().getOrderByUserId(
+          parseInt(metadata.metadata.userId)
+        );
+        await new OrderService().paymentRecieved(
+          prismaOrder!.id,
+          metadata.amount_total / 100
```

Comment:
the /100 operation should probably be in a dedicated function, as it's a stripe convention that amounts are in cents, maybe not the case everywhere